csrc/id_model/indexing.cpp

```diff
@@ -339,6 +372,27 @@ void TensorIndexer::buildLoopIndexMap() {
   }
 }
 
+bool TensorIndexer::shouldUseZeroIndex(const ValGroup& loop_group) const {
+  // All loops in this set are non-parallel, non-concretized broadcast
+  //  iterdomains, their "index variable" should be zero. This
+  //  condition should be included in the next triviality check, but
+  //  just checking isBroadcast should be more efficient.
+  if (std::all_of(loop_group->begin(), loop_group->end(), [](Val* val) {
+        return val->as<IterDomain>()->isBroadcast();
+      })) {
+    return true;
+  }
+
+  // Trivial loop
+  auto leaf_id =
+      getLoopPromotion(loop_group->front()->as<IterDomain>(), id_model_);
+  if (!leaf_id->maybePartial() && simplifyExpr(leaf_id->extent())->isOneInt()) {
```

Comment:
Do we support partial IterDomains?

Reply:
No. I think I just put this mostly by following what we have in `kir::ForLoop::isTrivial` before the shift removal. Now that it's removed, this seems more confusing than necessary. I'll remove it.